doc/hooks/versionCheckHook.section.md

```diff
@@ -16,20 +22,65 @@ stdenv.mkDerivation (finalAttrs: {
 
-It does so in a clean environment (using `env --ignore-environment`), and it checks for the `${version}` string in both the `stdout` and the `stderr` of the command. It will report to you in the build log the output it received and it will fail the build if it failed to find `${version}`.
+Below we tabulate the differences between `versionCheckHook` and `testers.testVersion`.
 
-The variables that this phase control are:
+| Item                                      | `versionCheckHook`                                                                                                                                                      | `testers.testVersion` via `passthru.tests.version`                                                                                           |
+|-------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
+| Customization                             | Besides the attributes described above, `versionCheckHook` provides no other methods for controlling it.                                                                | `tests.version` has access to the whole feature set of Nixpkgs, like multiple version test derivations and tools outside the package inputs. |
+| Execution time                            | `versionCheckHook` is executed automatically whenever the package is executed, e.g. `nix-build -A pkg`, since it is a phase executed during the package build time.     | `tests.version` is executed only when explicitly called, e.g. `nix-build -A pkg.tests.version`; it is executed after package build time.     |
+| Execution environment                     | `versionCheckHook` uses `env --ignore-environment --chdir=/` to run the executables in an environment as clean as possible, but there is no way to change its behavior. | `tests.version` executes in an environment identical to that of a consumer, independent from the package build environment.                  |
+| Rebuild after modification                | Modifying `versionCheckHook` triggers the package rebuild.                                                                                                              | `tests.version` does not rebuild the package, since it runs after package build time.                                                        |
+| Overhead during package build time        | Negligible. Although executed during the the package build time, `versionCheckHook` is lean and executes few commands.                                                  | Zero, since `tests.version` is a derivation that runs after package build time.                                                              |
```

Comment:
Can be more accurate:

```suggestion
| Overhead during package build time        | Negligible - executes the main package's command 1-2 times, during the package's build time. | Zero, since `tests.version` is an independent derivation. |
```

It's the 2nd time you write "a derivation that runs after a package build time" or something roughly the same - I hope you are OK with the fact I emphasize that not necessarily people will build (and not run) that derivation, because people might forget to do it by themselves or look at ofborg's attempts (which are [sometimes](https://github.com/NixOS/nixpkgs/pull/353607#issuecomment-2456283386) inaccurate).